### PR TITLE
Added settings for group chats

### LINKED ISSUE #272: Group chats do not call notify() to generate a platform specific notification.
It's very easy to miss group chat messages as there are never any notifications.

It looks like this is because compared to the [FRIEND_MESSAGE event handling](https://github.com/GrayHatter/uTox/blob/cea80f010eca32f0898b87e2e0efd65364db2018/src/tox.c#L1509) code, which calls [friend_addmessage](https://github.com/GrayHatter/uTox/blob/b80ace04297595198bb3f2865e75dbc069433e3b/src/friend.c#L242), the [GROUP_MESSAGE event handling code](https://github.com/GrayHatter/uTox/blob/cea80f010eca32f0898b87e2e0efd65364db2018/src/tox.c#L1627) is missing the section of code that also calls notify shown below.

``` cpp
/* if msg_type is text/action ? create tray popup */
switch(msg->msg_type) {
case MSG_TYPE_TEXT:
case MSG_TYPE_ACTION_TEXT: {
    char_t m[msg->length + 1];
    memcpy(m, msg->msg, msg->length);
    m[msg->length] = 0;
    notify(f->name, f->name_length, m, msg->length, f);
    break;
}
}
```
## <bountysource-plugin>

Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/32573306-group-chats-do-not-call-notify-to-generate-a-platform-specific-notification?utm_campaign=plugin&utm_content=tracker%2F21546874&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F21546874&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by endoffile78**
Supported settings:
* setting the topic
* setting the notification level (on, when mentioned, off)

I tested it on linux and windows. 

Closes #272

---
This change is [Reviewable](https://reviewable.io/reviews/grayhatter/utox/378)

### CODE REVIEW

File: src/roster.c [hunk 1]
```diff
@@ -930,7 +942,9 @@ static void contextmenu_list_onselect(uint8_t i) {
             case ITEM_GROUP: {
                 panel_group_chat.disabled = 0;
                 GROUPCHAT *g = right_mouse_item->data;
-                if (i == 0) {
+                if(i == 0){
```
Comment (tsudoko) [inline src/roster.c]: inconsistent spacing

File: src/roster.c [hunk 2]
```diff
@@ -930,7 +942,9 @@ static void contextmenu_list_onselect(uint8_t i) {
...
+                    roster_init_group_settings_page();
+                }else if (i == 1) {
```
Comment (tsudoko) [inline src/roster.c]: inconsistent spacing

File: src/tox_callbacks.c [hunk 3]
```diff
@@ -125,8 +125,12 @@ static void callback_group_invite(Tox *tox, int fid, uint8_t type, const uint8_t
 static void callback_group_message(Tox *tox, int gid, int pid, const uint8_t *message, uint16_t length,
                                    void *UNUSED(userdata))
 {
+    GROUPCHAT *g = &group[gid];
     debug_notice("Group Message (%u, %u): %.*s\n", gid, pid, length, message);
-    group_add_message(&group[gid], pid, message, length, MSG_TYPE_TEXT);
+    group_add_message(g, pid, message, length, MSG_TYPE_TEXT);
+    if(g->notify == 0 || (g->notify == 1 && strstr((const char *)message, (const char *)self.name) != NULL)){
```
Comment (tsudoko) [inline src/tox_callbacks.c]: inconsistent spacing

File: src/ui/ui_dropdown.c [hunk 4]
```diff
@@ -149,6 +149,13 @@ static void dropdown_friend_autoaccept_ft_onselect(const uint16_t i, const DROPD
     debug("Friend %u, is now accepting ft auto %u\n", f->number, i);
 }
 
+
+static void dropdown_notify_groupchats_onselect(const uint16_t i, const DROPDOWN* UNUSED(dm)){
```
Comment (tsudoko) [inline src/ui/ui_dropdown.c]: inconsistent spacing

File: src/ui/ui_edits.c [hunk 5]
```diff
@@ -485,6 +486,13 @@ static void edit_profile_password_update(EDIT *edit) {
     }
 }
 
+static void edit_group_topic_onenter(EDIT *edit){
```
Comment (tsudoko) [inline src/ui/ui_edits.c]: inconsistent spacing

**Comment by GrayHatter** [comment]
@endoffile78 for linux and windows, it should be as simple as modifying the notify() fxns.

Also, the notify settings aren't saved anywhere. I can add that in if you don't plan too.

**Comment by endoffile78** [comment]
@GrayHatter Yeah thats what I did. I didn't add in the ability to save the settings because it specific to the groupchat and since the groupchats aren't persistent it doesn't make sense to have that right now.

**Comment by GrayHatter** [comment]
I meant OSX and windows... 

Also, what do you think about having a default setting?

**Comment by endoffile78** [comment]
@GrayHatter Ok I'll change the function for OSX. I was thinking about that and I think that would be a good idea.

**Comment by endoffile78** [comment]
@GrayHatter I can look into saving the global setting later.

File: src/cocoa/interaction.m [hunk 6]
```diff
@@ -803,24 +803,27 @@ void notify(char_t *title, uint16_t title_length, const char_t *msg, uint16_t ms
...
-                   h = CGImageGetHeight(im->image) / im->scale;
-            NSImage *i = [[NSImage alloc] initWithCGImage:im->image size:(CGSize){w, h}];
-            if ([usernotification respondsToSelector:@selector(set_identityImage:)]) {
-                [usernotification set_identityImage:i];
-            } else {
-                NSLog(@"WARNING: OS X has broken the private api I use to set notification avatars. "
-                      "If you see this message please update uTox (if you're on latest, file a bug)");
-            }
-            if ([usernotification respondsToSelector:@selector(set_identityImageHasBorder:)]) {
+        if(!is_group){
```
Comment (tsudoko) [inline src/cocoa/interaction.m]: inconsistent spacing

File: src/roster.c [hunk 7]
```diff
@@ -942,9 +942,9 @@ static void contextmenu_list_onselect(uint8_t i) {
             case ITEM_GROUP: {
                 panel_group_chat.disabled = 0;
                 GROUPCHAT *g = right_mouse_item->data;
-                if(i == 0){
+                if (i == 0){
```
Comment (tsudoko) [inline src/roster.c]: inconsistent spacing (`){`)

**Comment by GrayHatter** [comment]
@endoffile78 merged with a patch, thanks mate!!

**Pull request merged**